commands/delete/nodepool/command.go

```diff
@@ -0,0 +1,216 @@
+
+	return Arguments{
+		APIEndpoint:       endpoint,
+		AuthToken:         token,
+		ClusterID:         parts[0],
+		Force:             flags.Force,
+		NodePoolID:        parts[1],
+		UserProvidedToken: flags.Token,
+		Verbose:           flags.Verbose,
+	}
+}
+
+func verifyPreconditions(args Arguments) error {
+	if args.AuthToken == "" && args.UserProvidedToken == "" {
+		return microerror.Mask(errors.NotLoggedInError)
+	}
+	if args.ClusterID == "" {
+		return microerror.Mask(errors.ClusterIDMissingError)
+	}
+	if args.NodePoolID == "" {
+		return microerror.Mask(errors.NodePoolIDMissingError)
+	}
+
+	return nil
+}
+
+func printValidation(cmd *cobra.Command, positionalArgs []string) {
+	var err error
+	args := collectArguments(positionalArgs)
+	if err == nil {
+		err = verifyPreconditions(args)
+	}
+
+	if err == nil {
+		return
+	}
+
+	errors.HandleCommonErrors(err)
+	client.HandleErrors(err)
+
+	headline := ""
+	subtext := ""
+
+	// print output
+	fmt.Println(color.RedString(headline))
+	if subtext != "" {
+		fmt.Println(subtext)
+	}
+	os.Exit(1)
+}
+
+// deleteNodePool is the business function sending our deletion request to the API
+// and returning true for success or an error.
+func deleteNodePool(args Arguments) (bool, error) {
+	// confirmation
+	if !args.Force {
+		question := fmt.Sprintf("Do you really want to delete node pool '%s' from cluster '%s'?", args.NodePoolID, args.ClusterID)
+		confirmed := confirm.Ask(question)
+		if !confirmed {
+			return false, nil
+		}
+	}
+
+	clientWrapper, err := client.NewWithConfig(args.APIEndpoint, args.UserProvidedToken)
+	if err != nil {
+		return false, microerror.Mask(err)
+	}
+
+	auxParams := clientWrapper.DefaultAuxiliaryParams()
+	auxParams.ActivityName = activityName
+
+	_, err = clientWrapper.DeleteNodePool(args.ClusterID, args.NodePoolID, auxParams)
+	if err != nil {
+		// return specific error types for the cases we care about most.
+		if clienterror.IsAccessForbiddenError(err) {
+			return false, microerror.Mask(errors.AccessForbiddenError)
+		}
+		if clienterror.IsNotFoundError(err) {
+			return false, microerror.Mask(errors.ClusterNotFoundError)
+		}
+
+		return false, microerror.Mask(err)
+	}
+
+	return true, nil
+}
+
+func printResult(cmd *cobra.Command, positionalArgs []string) {
+	args := collectArguments(positionalArgs)
+
+	deleted, err := deleteNodePool(args)
+	if err != nil {
+		errors.HandleCommonErrors(err)
+		client.HandleErrors(err)
+
+		headline := ""
+		subtext := ""
+
+		switch {
+		// If there are specific errors to handle, add them here.
+		default:
+			headline = err.Error()
+		}
+
+		// print output
+		fmt.Println(color.RedString(headline))
+		if subtext != "" {
+			fmt.Println(subtext)
+		}
+		os.Exit(1)
+	}
+
+	if deleted {
+		fmt.Println(color.GreenString("Node pool '%s' in cluster '%s' will be deleted as soon as all workloads are terminated.", args.NodePoolID, args.ClusterID))
+	} else if args.Verbose {
+		fmt.Println(color.WhiteString("Aborted."))
```

Comment:
So we get the full stop here now. Why is that FMI? I have been told single word labels, titles, responses do not need that.

Reply:
Probably you crept into my head and changed my thinking. Can't tell. Humans.